File: app/src/main/java/com/yes_u_du/zuyger/ui/reg_and_login_utils/forgot/ForgotPass.java
package com.yes_u_du.zuyger.ui.reg_and_login_utils.forgot;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.google.firebase.auth.FirebaseAuth;
import com.yes_u_du.zuyger.R;
import com.yes_u_du.zuyger.ui.reg_and_login_utils.auth.LoginFragment;

public class ForgotPass extends AppCompatActivity {
    Button fr_btn;
    EditText reset_email;

    private FirebaseAuth mAuth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_forgot_pass);
        mAuth = FirebaseAuth.getInstance();

        reset_email = findViewById(R.id.reset_email);

        fr_btn = findViewById(R.id.fr_btn);
        fr_btn.setOnClickListener(v -> {
            String userEmail = reset_email.getText().toString();

            if (TextUtils.isEmpty(userEmail)) {
                Toast.makeText(ForgotPass.this, "Поля пусты", Toast.LENGTH_LONG).show();
            } else {
                mAuth.sendPasswordResetEmail(userEmail).addOnCompleteListener(task -> {
                    if (task.isSuccessful()) {
                        Toast.makeText(ForgotPass.this, "Ссылка для восстановления пароля отправлена на ваш email", Toast.LENGTH_LONG).show();
                        startActivity(new Intent(ForgotPass.this, LoginFragment.class));
                        finish();
                    } else {
                        Toast.makeText(ForgotPass.this, "Ошибка", Toast.LENGTH_LONG).show();
                    }
                });
            }
        });
    }
}

File: app/src/main/java/com/yes_u_du/zuyger/ui/reg_and_login_utils/forgot/ForgotPass.kt
package com.yes_u_du.zuyger.ui.reg_and_login_utils.forgot

import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.android.gms.tasks.Task
import com.google.firebase.auth.FirebaseAuth
import com.yes_u_du.zuyger.R
import com.yes_u_du.zuyger.databinding.ActivityForgotPassBinding
import com.yes_u_du.zuyger.ui.reg_and_login_utils.auth.LoginFragment

class ForgotPass : AppCompatActivity() {
	
	private lateinit var binding: ActivityForgotPassBinding
	private var mAuth: FirebaseAuth? = null
	
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		binding = ActivityForgotPassBinding.inflate(layoutInflater)
		setContentView(binding.root)
		mAuth = FirebaseAuth.getInstance()
		
		binding.frBtn.setOnClickListener {
			val userEmail = binding.resetEmail.editText!!.text.toString()
			if (TextUtils.isEmpty(userEmail)) {
				Toast.makeText(this@ForgotPass,
					resources.getString(R.string.empty_fields),
					Toast.LENGTH_LONG).show()
			} else {
				mAuth!!.sendPasswordResetEmail(userEmail)
					.addOnCompleteListener { task: Task<Void?> ->
						if (task.isSuccessful) {
							Toast.makeText(this@ForgotPass,
								resources.getString(R.string.email_info_sent),
								Toast.LENGTH_LONG).show()
							startActivity(Intent(this@ForgotPass,
								LoginFragment::class.java))
							finish()
						} else {
							Toast.makeText(this@ForgotPass,
								resources.getString(R.string.error),
								Toast.LENGTH_LONG).show()
						}
					}
			}
		}
	}
}